src/main/java/seedu/address/logic/commands/ingredientcommands/IngredientResetAllCommand.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.logic.commands.ingredientcommands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ingredient.Amount;
+import seedu.address.model.ingredient.Ingredient;
+
+/**
+ * Set the level of one specific ingredient to a specific level.
+ */
+public class IngredientResetAllCommand extends Command {
+
+    public static final String COMMAND_WORD = "i-reset-all";
+    public static final String RESET_AMOUNT = "0";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " :reset all the ingredient levels in tCheck."
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "All ingredient levels are reset to 0.";
+    public static final String MESSAGE_NO_CHANGE = "All Ingredient levels are already at 0.";
+
+    /**
+     * Constructs an IngredientResetAll command.
+     */
+    public IngredientResetAllCommand() {
+        super();
+    }
+
+    /**
+     * Executes the command and returns the result message.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return feedback message of the operation result for display
+     * @throws CommandException If an error occurs during command execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Ingredient> lastShownList = model.getFilteredIngredientList();
+
+        boolean isNotAlreadyReset = false;
+        Amount resetAmount = new Amount(RESET_AMOUNT);
+        for (Ingredient i : lastShownList) {
+            if (!i.getAmount().equals(resetAmount)) {
+                isNotAlreadyReset = true;
+                break;
+            }
+        }
+
+        if (!isNotAlreadyReset) {
+            throw new CommandException(MESSAGE_NO_CHANGE);
+        }
+
+        for (Ingredient i : lastShownList) {
+            Ingredient ingredientToEdit = lastShownList.get(lastShownList.indexOf(i));
+            Ingredient updatedIngredient = createResetIngredient(i);
+            model.setIngredient(ingredientToEdit, updatedIngredient);
+        }
+        return new CommandResult(String.format(MESSAGE_SUCCESS));
```

Comment:
No need to use `String.format` if there is nothing to format.

Reply:
Noted with thanks